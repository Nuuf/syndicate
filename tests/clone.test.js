/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('clones an entity', () => {
  const root = syndicate.createRoot();
  syndicate.add(root, syndicate.create({}, 'parent'));
  syndicate.add(root, syndicate.create({ someData: 'hello world' }, 'child'), syndicate.find(root, 'parent'));
  syndicate.clone(root, syndicate.find(root, 'child'), syndicate.START);
  syndicate.clone(root, syndicate.find(root, 'child'), syndicate.END);
  syndicate.clone(root, syndicate.find(root, 'child'), syndicate.START);
  syndicate.clone(root, syndicate.find(root, 'child'), syndicate.END);
  syndicate.clone(root, syndicate.find(root, 'child'), syndicate.START);
  expect(JSON.stringify(root, null, 2)).toMatchSnapshot();
});
