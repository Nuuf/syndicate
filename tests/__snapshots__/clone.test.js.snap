// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clones an entity 1`] = `
"[
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":-1,\\\\\\"parentId\\\\\\":null,\\\\\\"index\\\\\\":0,\\\\\\"id\\\\\\":\\\\\\"root\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"root\\\\\\"},\\\\\\"childrenIndices\\\\\\":[1],\\\\\\"childrenIds\\\\\\":[\\\\\\"parent\\\\\\"]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":1,\\\\\\"id\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[2,3,4,5,6,7],\\\\\\"childrenIds\\\\\\":[\\\\\\"child\\\\\\",\\\\\\"1\\\\\\",\\\\\\"2\\\\\\",\\\\\\"3\\\\\\",\\\\\\"4\\\\\\",\\\\\\"5\\\\\\"]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":2,\\\\\\"id\\\\\\":\\\\\\"child\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":\\\\\\"hello world\\\\\\"},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":3,\\\\\\"id\\\\\\":\\\\\\"1\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":\\\\\\"hello world\\\\\\"},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":4,\\\\\\"id\\\\\\":\\\\\\"2\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":\\\\\\"hello world\\\\\\"},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":5,\\\\\\"id\\\\\\":\\\\\\"3\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":\\\\\\"hello world\\\\\\"},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":6,\\\\\\"id\\\\\\":\\\\\\"4\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":\\\\\\"hello world\\\\\\"},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":7,\\\\\\"id\\\\\\":\\\\\\"5\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"someData\\\\\\":\\\\\\"hello world\\\\\\"},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\"
]"
`;
